### PR TITLE
Bring in G5 Updatable

### PR DESCRIPTION
**Pull request by jcstringer**
https://www.pivotaltracker.com/story/show/94445022

This one's gonna need a lot of QA and peer review. Master has been merged in as of 5/27/15.

### CODE REVIEW

File: app/controllers/api/v1/locations_controller.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Api::V1::LocationsController < Api::V1::ApplicationController
   def index
-    render json: Location.all
+    render json: G5Updatable::Location.all, each_serializer: LocationSerializer
```
Comment (thescubageek) [inline app/controllers/api/v1/locations_controller.rb]: nice, never seen each_serializer

File: app/decorators/location_decorator.rb [hunk 2]
```diff
@@ -10,6 +10,10 @@ class LocationDecorator < Draper::Decorator
     :service_1, :service_2, :service_3, :service_4, :service_5,
     :website
 
+  def self.object_class
```
Comment (thescubageek) [inline app/decorators/location_decorator.rb]: What does this do?

File: app/models/clients/multi_domain_client.rb [hunk 3]
```diff
@@ -1,5 +0,0 @@
-class MultiDomainClient < Client
```
Comment (thescubageek) [inline app/models/clients/multi_domain_client.rb]: Nice!

File: app/models/concerns/after_update_set_path_settings.rb [hunk 4]
```diff
@@ -8,7 +9,8 @@ module AfterUpdateSetPathSettings
   private
 
   def should_update_location_paths?
-    city_changed? || state_changed?
+    observed_properties = %w{city state}
```
Comment (thescubageek) [inline app/models/concerns/after_update_set_path_settings.rb]: Much better

File: app/models/concerns/after_update_set_setting_locations_navigation.rb [hunk 5]
```diff
@@ -9,11 +10,12 @@ module AfterUpdateSetSettingLocationsNavigation
   private
 
   def should_update_locations_navigation_settings?
-    state_changed? || city_changed? || name_changed?
+    observed_properties = %w{city state name}
+    properties_changed? && property_changed?(observed_properties)
```
Comment (thescubageek) [inline app/models/concerns/after_update_set_setting_locations_navigation.rb]: Any way we can DRY up the three repetitions of the follow?:
```
properties_changed? && property_changed?(observed_properties)
```

File: app/decorators/location_decorator.rb (hunk 2 above)
Comment (jcstringer) [inline app/decorators/location_decorator.rb]: The decorator assumes it should use a class of the same name, in this case we want it to actually use G5Updatable::Location. You can over-ride the method per decorator to do this.

File: app/models/concerns/location_liquid_parameters.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@ module LocationLiquidParameters
 
   def location_site_liquid(l_location)
     site_info = {
-      "website_urn"               => l_location.website.urn,
+      "website_urn"               => l_location.reload.website.urn,
```
Comment (thescubageek) [inline app/models/concerns/location_liquid_parameters.rb]: Why is this reload necessary?

File: lib/color_finder.rb [hunk 7]
```diff
@@ -35,6 +35,6 @@ def web_theme
   end
 
   def valid?
-    @location && @location.website && @location.website.website_template
+    @location.try(:reload) && @location.reload.website && @location.website.website_template
```
Comment (thescubageek) [inline lib/color_finder.rb]: We need these reloads because...?

**Comment by jcstringer** [comment]
@thescubageek the .htaccess specs are commented out at the advice of @levibrown who has already redone these specs in another PR that will be merged in soonish. So rather then fix them I'm moving them out of the way, they were writing to the file system multiple times on every run.

**Pull request merged**